LambdaEngine/Source/Rendering/RenderGraphParser.cpp

```diff
@@ -191,6 +203,43 @@ namespace LambdaEngine
 			orderedPipelineStages.PushBack({ ERenderGraphPipelineStageType::SYNCHRONIZATION, uint32(orderedSynchronizationStages.GetSize()) - 1 });
 		}
 
+		if (generateLineRendererStage)
+		{
+			SynchronizationStageDesc lineRendererSynchronizationStage = {};
+
+			{
+				//This is just a dummy as it will be removed in a later stage
+				RenderGraphResourceSynchronizationDesc resourceSynchronization = {};
+				resourceSynchronization.PrevRenderStage = RENDER_GRAPH_LINE_RENDERER_STAGE_NAME;
+				resourceSynchronization.NextRenderStage = generateImGuiStage ? RENDER_GRAPH_IMGUI_STAGE_NAME : "PRESENT (Not a Render Stage)";
+				resourceSynchronization.PrevBindingType = ERenderGraphResourceBindingType::ATTACHMENT;
+				resourceSynchronization.NextBindingType = generateImGuiStage ? ERenderGraphResourceBindingType::ATTACHMENT : ERenderGraphResourceBindingType::PRESENT;
+				resourceSynchronization.PrevQueue		= ECommandQueueType::COMMAND_QUEUE_TYPE_GRAPHICS;
+				resourceSynchronization.NextQueue		= ECommandQueueType::COMMAND_QUEUE_TYPE_GRAPHICS;
+				resourceSynchronization.ResourceName	= RENDER_GRAPH_BACK_BUFFER_ATTACHMENT;
+				resourceSynchronization.ResourceType	= ERenderGraphResourceType::TEXTURE;
```

Comment:
Can you give context to this? Why will it be removed?

Reply:
This is added here in order to create the transistion of the buffer. However it will be removed later on as the render pass will handle this transistion instead. This is done in a later stage in which all of the barriers are checked. Without the addition of the synchronization here no check will be made for this buffer and therefore no transistion will occur, which might create problems.